lib/packwerk/cli.rb

```diff
@@ -181,6 +182,35 @@ def check(paths)
       all_offenses.empty?
     end
 
+    def detect_stale_violations(paths,
```

Comment:
This class is getting too big and complicated to maintain. I think it would be good to apply the [scout rule](https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html) here, leaving a class cleaner than you found it. 

I would suggest moving your changes to a separate class so we can stop the rot. This should also be pretty straightforward since it doesn't require a big refactor.

Reply:
I think it makes sense for the `CLI` class to have one method per packwerk subcommand, however we can consider reducing the size of this method considerably by moving most of its implementation somewhere else.

IMO, `CLI` should just be a thin wrapper that does the CLI specific stuff (like managing the output stream and parsing parameters) and delegates to other classes for everything else.

Reply:
I agree that `CLI` is a mess and should be reorganised but am more ambivalent about whether it necessarily must happen in this PR. Could one of us open a new PR to do the cleanup, either before or after this one lands?

Reply:
I agree with Tom - Actually that was what I had in mind but failed to address the point. I meant to say "what should happen to CLI is a larger refactor and should thus be done separately"